waldo/tool/flutter/flutter.go

```diff
@@ -42,27 +40,22 @@ func MakeFlutterBuilder(absPath, relPath string, verbose bool, ios *lib.IOStream
 
 	ios.Printf("\nFinding all supported build flavors…\n")
 
-	name, err := detectAppName(absPath)
+	fi, err := DetectFlutterInfo(absPath, platform, ios)
 
 	if err != nil {
 		return nil, "", lib.PlatformUnknown, err
 	}
 
-	flavors, err := detectFlavors(absPath, platform, ios)
+	flavor, err := determineFlavor(fi.Flavors, verbose, ios)
 
 	if err != nil {
 		return nil, "", lib.PlatformUnknown, err
 	}
 
-	flavor, err := determineFlavor(flavors, verbose, ios)
-
-	if err != nil {
-		return nil, "", lib.PlatformUnknown, err
-	}
-
-	fb := newFlutterBuilder(flavor)
+	fb := &FlutterBuilder{
```

Comment:
nit: weird go styling here with '{' and '}' sticking to the rest without space

Reply:
Would you still think it weird if there were more than one argument?